File: java/TwoPointer/Container_With_Most_Water_11.java
package java.TwoPointer;

import java.util.*;
import java.io.*;

// import static java.lang.Math.max;
// import static java.lang.Math.min;
// import static java.lang.Math.abs;
// import static java.lang.System.out;

public class Container_With_Most_Water_11 {

    public int maxArea(int[] nums) {
        int i, j, l, r, ans;
        i = l = ans = 0;
        j = r = nums.length - 1;
        while (i < j) {
            ans = Math.max(ans, Math.min(nums[i], nums[j]) * (j - i));
            if (nums[i] < nums[j])
                ++i;
            else if (nums[i] > nums[j])
                --j;
            else {
                l = i;
                r = j;
                while (l < r) {
                    if (l - r != 0)
                        break;
                    ++l;
                    --r;
                }
                if (nums[l] > nums[r])
                    --j;
                else
                    ++i;
            }
        }
        return ans;

    }

    public static void main(String[] args) throws FileNotFoundException {
        File f = new File("output.txt");

        // Scanner ip = new Scanner(System.in);
        try (Scanner ip = new Scanner(new File("input.txt"));
                PrintWriter op = new PrintWriter(f);) {
            Container_With_Most_Water_11 obj = new Container_With_Most_Water_11();
            int grid[] = { 1, 8, 6, 2, 5, 4, 8, 3, 7 };
            op.println(obj.maxArea(grid));

        }
    }

}

File: Code/TwoPointer/Container_With_Most_Water_11.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int maxArea(vector<int> nums)
    {
        int i, j, l, r, ans;
        i = l = ans = 0;
        j = r = nums.size() - 1;
        while (i < j)
        {
            ans = max(ans, min(nums[i], nums[j]) * (j - i));
            if (nums[i] < nums[j])
                ++i;
            else if (nums[i] > nums[j])
                --j;
            else
            {
                l = i;
                r = j;
                while (l < r)
                {
                    if (l - r != 0)
                        break;
                    ++l;
                    --r;
                }
                if (nums[l] > nums[r])
                    --j;
                else
                    ++i;
            }
        }
        return ans;
    }
};
int main()
{
#ifndef ONLINE_JUDGE
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
#endif
    Solution obj;
    auto ans = obj.maxArea({1, 8, 6, 2, 5, 4, 8, 3, 7});
    // cout << ans;
    return 0;
}